pkg/log/log.go

```diff
@@ -1,47 +1,90 @@
 package log
 
 import (
-	"github.com/sirupsen/logrus"
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"log/slog"
+	"os"
 )
 
 // Logger is the main struct that any athens
 // internal service should use to communicate things.
 type Logger struct {
-	*logrus.Logger
+	*slog.Logger
+
+	Out *bytes.Buffer
 }
 
 // New constructs a new logger based on the
 // environment and the cloud platform it is
 // running on. TODO: take cloud arg and env
 // to construct the correct JSON formatter.
-func New(cloudProvider string, level logrus.Level, format string) *Logger {
-	l := logrus.New()
+func New(cloudProvider string, level slog.Level, format string) *Logger {
+	l := slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: level}))
 	switch cloudProvider {
 	case "GCP":
-		l.Formatter = getGCPFormatter()
+		l = getGCPFormatter(level)
 	default:
-		l.Formatter = parseFormat(format)
+		l = parseFormat(format, level)
 	}
-	l.Level = level
+	slog.SetDefault(l)
 	return &Logger{Logger: l}
 }
 
 // SystemErr Entry implementation.
 func (l *Logger) SystemErr(err error) {
-	e := &entry{Entry: logrus.NewEntry(l.Logger)}
+	e := &entry{l.Logger}
 	e.SystemErr(err)
 }
 
 // WithFields Entry implementation.
 func (l *Logger) WithFields(fields map[string]any) Entry {
-	e := l.Logger.WithFields(fields)
+	attrs := make([]any, 0, len(fields))
+	for k, v := range fields {
+		attrs = append(attrs, slog.Any(k, v))
+	}
+	return &entry{logger: l.Logger.With(attrs...)}
+}
+
+func (l *Logger) WithField(key string, value any) Entry {
+	keys := map[string]any{
+		key: value,
+	}
+	return l.WithFields(keys)
+}
+
+func (l *Logger) WithError(err error) Entry {
+	keys := map[string]any{
+		"error": err,
+	}
+	return l.WithFields(keys)
+}
+
+func (l *Logger) WithContext(ctx context.Context) Entry {
+	keys := map[string]any{
+		"context": ctx,
+	}
+	return l.WithFields(keys)
+}
 
-	return &entry{e}
+// Define WriterLevel
+func (l *Logger) WriterLevel(level slog.Level) *io.PipeWriter {
```

Comment:
Please let me know of this function implementation?

had to take help of AI here.